include/lapack/potrf.hpp

```diff
@@ -81,22 +82,19 @@ int potrf( uplo_t uplo, matrix_t& A, opts_t&& opts )
     // Constants
     const real_t one( 1.0 );
     const idx_t n  = nrows(A);
-
-    // Options
-    const idx_t nb = get_nb(opts);
+    const idx_t nb = opts.nb;
 
     // check arguments
-    lapack_error_if(    uplo != Uplo::Lower &&
-                        uplo != Uplo::Upper, -1 );
-    lapack_error_if(    access_denied( uplo, write_policy(A) ), -1 );
-    lapack_error_if(    nrows(A) != ncols(A), -2 );
+    tlapack_check_param(  uplo == Uplo::Lower || uplo == Uplo::Upper, -1 );
+    tlapack_check_access( access_granted( uplo, write_policy(A) ),    -2 );
+    tlapack_check_sizes(  nrows(A) == ncols(A),                       -2 );
```

Comment:
These changes look good to me.

Reply:
Hi. They look good, but I have been thinking more on something like

```c++
tlapack_check( opts.paramCheck, uplo == Uplo::Lower || uplo == Uplo::Upper, -1 );
```

or maybe

```c++
tlapack_check( err.param, uplo == Uplo::Lower || uplo == Uplo::Upper, -1 );
```

This will expose what flag is being checked at runtime.